### ISSUE TITLE
Games in SD will not start when  NAND Emulation is set to Game save or Full

### ISSUE THREAD

**Issue by TeenTin** [Fledge68/WiiFlow_Lite#196]
Games in **SD** will **not start** when **NAND Emulation** is set to **Game save** or **Full**.

When pressing Start, WiiFlow Lite tries to load the game but fail, returning to the WiiFlow menu.  Any fix ?

**Comment by Fledge68** [comment]
wii games? or emunand games?

**Comment by TeenTin** [comment]
Wii games

**Comment by Fledge68** [comment]
did you set up an emu nand on your sd card in a subfolder of sd:/nands? example - sd:/nands/mynand

did you go to main settings and emu nand settings to set up your emu nand as a saves nand?

also some wii games do not work with gamesaves on emu nand.

**Comment by TeenTin** [comment]
> did you set up an emu nand on your sd card in a subfolder of sd:/nands? example - sd:/nands/mynand
> 

Yes, I did.

> did you go to main settings and emu nand settings to set up your emu nand as a saves nand?
> 

Yes, I did.

> also some wii games do not work with gamesaves on emu nand.

I have tested tons of games, none works.  However, when I load games from USB instead of SD, almost all games work.

**Comment by gingerbeardman** [comment]
Did you try a different SD card, or the same SD card formatted to the correct guidelines (FAT32 etc)?

**Comment by TeenTin** [comment]
> Did you try a different SD card, or the same SD card formatted to the correct guidelines (FAT32 etc)?

I have tried different SD cards (different sizes and brands), all formatted to FAT32 (32 kB cluster).

**Comment by gingerbeardman** [comment]
Wii or vWii?

Please check your setup using [ModMii](https://gbatemp.net/threads/modmii-for-windows-official-support-topic.207126/)

**Comment by Fledge68** [comment]
so when you launch a game - after the animated wiiflow logo do you see a white bar progress across the bottom of the screen and does it appear to go all the way or only part way?

**Comment by TeenTin** [comment]
> so when you launch a game - after the animated wiiflow logo do you see a white bar progress across the bottom of the screen and does it appear to go all the way or only part way?

No white bar at all.  After the animated wiiflow logo, I just got a black screen and then return to wiiflow automatically after some time.

**Comment by TeenTin** [comment]
> Wii or vWii?
> 
Wii

> Please check your setup using [ModMii](https://gbatemp.net/threads/modmii-for-windows-official-support-topic.207126/) and [SysCheck](https://gbatemp.net/threads/syscheck-hacksden-edition.356133/)

Wii check it soon.

**Comment by TeenTin** [comment]
May I ask if there is anybody who has successfully load games in SD with NAND Emulation set to Game save ?

**Comment by TeenTin** [comment]
> Please check your setup using [ModMii](https://gbatemp.net/threads/modmii-for-windows-official-support-topic.207126/) and [SysCheck](https://gbatemp.net/threads/syscheck-hacksden-edition.356133/)

Here is the file generated by SysCheck :
https://www.dropbox.com/s/1pcw5te4t18uhrb/sysCheck.csv?dl=0

P.S. I don't know how to use ModMii to check the setup, sorry.

**Comment by gingerbeardman** [comment]
> P.S. I don't know how to use ModMii to check the setup, sorry.

details are in this topic: https://gbatemp.net/threads/guide-to-safely-update-your-softmod-fixes-most-compatibility-issues.394397/

**Comment by TeenTin** [comment]
> > P.S. I don't know how to use ModMii to check the setup, sorry.
> 
> detailed "how to" is in this topic: https://gbatemp.net/threads/guide-to-safely-update-your-softmod-fixes-most-compatibility-issues.394397/


Tried.  ModMii freezes in the step of downloading HackMii Installer.
https://imgur.com/MUNKZmR
https://imgur.com/3ASqqsY

P.S. May I ask again : Is there anybody who has successfully load games in SD with NAND Emulation set to Game save ?

**Comment by gingerbeardman** [comment]
> Tried. ModMii freezes in the step of downloading HackMii Installer.
> https://imgur.com/MUNKZmR
> https://imgur.com/3ASqqsY

You don't need to install or download anything, just run the syscheck part of ModMii

> > At your computer you'll need to start up ModMii. ModMii will use your recently generated "sysCheck.csv" file to inform you on which of your IOS's/cIOS's need to be updated. I

> P.S. May I ask again : Is there anybody who has successfully load games in SD with NAND Emulation set to Game save ?

You will reach most users in the official support thread: https://gbatemp.net/threads/wiiflow-lite.422685/

**Comment by TeenTin** [comment]
> You don't need to install or download anything, just run the syscheck part of ModMii
> 
> > > At your computer you'll need to start up ModMii. ModMii will use your recently generated "sysCheck.csv" file to inform you on which of your IOS's/cIOS's need to be updated.
> 
> That's it!
> 

I have tried the IOS/cIOS suggested by ModMii.  They did not help.  Same result.

**Comment by Fledge68** [comment]
So did you go to emu nand settings page 3 and set your saves nand partition to SD?

**Comment by TeenTin** [comment]
> So did you go to emu nand settings page 3 and set your saves nand partition to SD?

Of course, I did.  Actually, I can play the virtual console games in the emu nand without problems.

**Comment by Fledge68** [comment]
so when you use your sd card you can set wiiflow to output a debug log to sd.

open wiiflow_lite.ini in a text editor (notepad++) and under [DEBUG] set sd_write_log=yes and save. run wiiflow and try to launch a wii game with emunand gamesave. exit wiiflow if you need to. now on the root of the sd card should be a file named wiiflow.log

copy and paste it here and maybe it will tell me what's going on.

**Comment by TeenTin** [comment]
> so when you use your sd card you can set wiiflow to output a debug log to sd.
> 
> open wiiflow_lite.ini in a text editor (notepad++) and under [DEBUG] set sd_write_log=yes and save. run wiiflow and try to launch a wii game with emunand gamesave. exit wiiflow if you need to. now on the root of the sd card should be a file named wiiflow.log
> 
> copy and paste it here and maybe it will tell me what's going on.

Thanks for your help.  Here is the content of the log file :

Welcome to WiiFlow WFL 5.4.6!
This is the debug output.
Real Wii
AHBPROT disabled = yes
patched isfs_permissions 1 times.
patched es_setuid 1 times.
patched es_identify 1 times.
patched hash_check 0 times.
patched new_hash_check 1 times.
Init ISFS
Found WiiFlow Save
Loading IOS Settings from wiiflow save
Deinit ISFS
AHBPROT disabled after IOS Reload: yes
patched isfs_permissions 0 times.
Init ISFS
D2X IOS249[56] v10
FAT Partition at sd: mounted.
using SD only, no need for USB mounting.
Running sound thread
Wiiflow boot.dol Location: sd:/apps/wiiflow
Data Directory: sd:/wiiflow
Wii Games Directory: %s:/wbfs
Nintendont found
GameCube Games Directory: %s:/games
Loading cIOS map
Found cIOS in slot 222
Found cIOS in slot 223
Found cIOS in slot 224
Found d2x base 56 in slot 249
Found d2x base 57 in slot 250
Found d2x base 58 in slot 251
Setting Wii games partition to: 0
emu nand path = sd:/nands/pln2o
saves nand path = sd:/nands/pln2o
Bootup completed!
Creating Gamelist
Adding wii list
Games found: 94
Displaying covers
Update Play log
Launching game RSPE01
Game ID RSPE01 requested IOS 9.
User selected 0
Changed requested IOS to 249.
Return to channel DWFA using d2x succeeded.
 
Welcome to WiiFlow WFL 5.4.6!
This is the debug output.
Real Wii
AHBPROT disabled = yes
patched isfs_permissions 1 times.
patched es_setuid 1 times.
patched es_identify 1 times.
patched hash_check 0 times.
patched new_hash_check 1 times.
Init ISFS
Found WiiFlow Save
Loading IOS Settings from wiiflow save
Deinit ISFS
AHBPROT disabled after IOS Reload: yes
patched isfs_permissions 0 times.
Init ISFS
D2X IOS249[56] v10
FAT Partition at sd: mounted.
using SD only, no need for USB mounting.
Running sound thread
Wiiflow boot.dol Location: sd:/apps/wiiflow
Data Directory: sd:/wiiflow
Wii Games Directory: %s:/wbfs
Nintendont found
GameCube Games Directory: %s:/games
Loading cIOS map
Found cIOS in slot 222
Found cIOS in slot 223
Found cIOS in slot 224
Found d2x base 56 in slot 249
Found d2x base 57 in slot 250
Found d2x base 58 in slot 251
Setting Wii games partition to: 0
emu nand path = sd:/nands/pln2o
saves nand path = sd:/nands/pln2o
Bootup completed!
Creating Gamelist
Adding wii list
Games found: 94
Displaying covers
Cleanup Coverflow
Stopping sound thread
Stopped sound thread
MEM1_freesize(): 8216560
MEM2_freesize(): 41865728

**Comment by Fledge68** [comment]
well that helped a little but still need to do more digging

**Comment by TeenTin** [comment]
May be the issue is due to the cIOS ?

**Comment by Fledge68** [comment]
possible. you may need to go in the game's settings and set IOS to a different cios.

I looked thru the code this morning and i can't find anything to cause your issue.
the debug log you did cuts off right before nand emulation is enabled. but that doesn't necessarily mean that is the problem. but it does mean shortly after that the debug output is unable to send more info to the sd card.

sorry but i have no solution to this issue.

**Comment by Fledge68** [comment]
update on this issue:

Blackbox and I have discovered that this is a real issue. game and emunand gamesave both on SD cause blackscreen crash. It seems to be a d2x cios issue involving mounting SD for emunand but not allowing wiiflow or any loader to remount SD for the game. looking into a solution but i wouldn't count on it.

**Comment by eku** [comment]
You may want to check [cIOs source](https://github.com/davebaol/d2x-cios) and fix it.

**Comment by gingerbeardman** [comment]
> You may want to check [cIOs source](https://github.com/davebaol/d2x-cios) and fix it.

It's unlikely the problem is in CIOS given that it hasn't been changed in 6 years. So I suppose a workaround is needed in WFL.

**Comment by Fledge68** [comment]
yes blackb0x and i have been looking at d2x source code but no clear fix. Ive given up but blackb0x is still looking into it.

**Comment by eku** [comment]
Is any progress being made on this issue? I have a second Wii here that I would like to run completely without external media and only with internal.

**Comment by eku** [comment]
> It's unlikely the problem is recent in CIOS given that it's source code hasn't been changed in 6 years. Also expecting users to update their d2x is too big an ask.

Well, [it has been updated 11 day ago](https://github.com/davebaol/d2x-cios/releases/tag/d2x-v10-b52-ST-FM0_5_0). Maybe [@xflak](https://github.com/xflak) could fix this issue for us.

**Comment by TeenTin** [comment]
> Having the same issues with the latest versions, workaround for me is to use [v5.4.9](https://github.com/Fledge68/WiiFlow_Lite/releases/tag/v5.4.9) which loads Wii games directly from SD no problem (with Force Load cIOS set to 249 or 250) as well as Emunand (with Boot Apploader on).

So, you mean the latest version v5.5.3 is not working with your workaround ?  Besides, would you please explain what is "Boot Apploader" and how to enable it ?  Thanks a lot !

**Comment by TeenTin** [comment]
> > So, you mean the latest version v5.5.3 is not working with your workaround ?
> 
> Yes same goes for v5.5.4 beta 3 ([5415c95](https://github.com/Fledge68/WiiFlow_Lite/commit/5415c95fefa44dba79e0a6e88f25b4a7a3766bf1)), loading Wii or Emunand games from SD will show a black screen and then return to WiiFlow menu on these newer versions.
> 
> > Besides, would you please explain what is "Boot Apploader" and how to enable it ?
> 
> For Emunand, "Boot Apploader" shows up in the game settings (which sets `apploader=yes` for that game in `gameconfig2.ini`).

I think you have misunderstood this issue.  This issue is :

"Games in SD will not start when NAND Emulation is set to Game save or Full."

Note :
1.  I can start Wii games from SD using v5.5.3, no problem.
2.  I can start Emunand games from SD using v5.5.3, no problem.
3.  However, when I start a Wii game from SD, while USING THE GAME SAVE from Emunand, I got a black screen.

**Comment by eku** [comment]
I tried the same scenario with `USB Loader GX` and there comes a notice that games and at the same time EmuNAND are not supported from SD card. The games start but the support for EmuNAND is disabled.

The restriction comes from the cIOS.

**Comment by mman876** [comment]
I think this issue applies to USB as well. I am using my wiiflow emunand on USB HDD, the same drive that I boot my wbfs games from. I have a few virtual console channels on the emunand as well as wii saves and miis. I can see the miis and run the channels fine through emunand, but if I launch a wii game with the saves emunand set to game or full, the game will not launch and it boots back to System Menu.

What I did, was I copied my nands folder from the usb to the sd. I set the emunand saves partition to SD, and I am now able to run some wii games and save them to emunand. I tried it with a few games so far.

New Super Mario Bros Wii: boots fine, I can see the saves in the emunand
Mario kart Wii: it boots, but I can't see any saves
Wii Sports: Boots, I can see saves and miis from emunand

Should this be considered part of the same issue, or maybe I should open another issue?

**Comment by mman876** [comment]
> I think this issue applies to USB as well. I am using my wiiflow emunand on USB HDD, the same drive that I boot my wbfs games from. I have a few virtual console channels on the emunand as well as wii saves and miis. I can see the miis and run the channels fine through emunand, but if I launch a wii game with the saves emunand set to game or full, the game will not launch and it boots back to System Menu.
> 
> What I did, was I copied my nands folder from the usb to the sd. I set the emunand saves partition to SD, and I am now able to run some wii games and save them to emunand. I tried it with a few games so far.
> 
> New Super Mario Bros Wii: boots fine, I can see the saves in the emunand Mario kart Wii: it boots, but I can't see any saves Wii Sports: Boots, I can see saves and miis from emunand
> 
> Should this be considered part of the same issue, or maybe I should open another issue?

My mistake, I checked my hdd and noticed it was on GPT instead of MBR. I fixed it, and now I can run usb games with emunand on usb! I have to say that this is very convenient, I can swap hdd's between wiis now and take all my saves and channels with me.

**Comment by TeenTin** [comment]
Why is this closed?

Fledge68 ***@***.***> 於 2025年8月27日 週三 上午6:37寫道：

**Comment by digimancer** [comment]
2026, same issue.
EmuNand + EmuSaves works perfectly in USB Loader GX running from SD

WiiFlow just no go... 

Only solution I can see to avoid having saves in multiplelocations, save games to real-nand... Down side is, can't move saves between consoles easily, plus side, when my 500gig fat32 eventually vomits, all of my saves are still safe... xD

**Comment by digimancer** [comment]
Again, only an issue in Wiiflow, emunand saves are working in ULGX. 

![Image](https://github.com/user-attachments/assets/f35b707f-5672-42f9-a4c7-8fb246b100e5)
![Image](https://github.com/user-attachments/assets/8af10667-bac8-409a-9d80-d2b345885dcc)
![Image](https://github.com/user-attachments/assets/5ae5df5d-5948-4bdb-b07f-453a8a4a0f79)

**Comment by digimancer** [comment]
The bright side of using real-nand though, is that I can lock my SD card now, and hopefully prevent my card from getting corrupted!  XD
